app/main.py

```diff
@@ -1,25 +1,66 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.private_name = f"_{name}"
+
+    def __get__(self, instance: Any, owner: Any) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError
```

Comment:
The `TypeError` raised here lacks a descriptive message. Consider adding a message to indicate that the value must be an integer.